File: app/src/Location.kt
import kotlin.math.pow
import kotlin.math.sqrt

data class Location(val x: Double, val y: Double)

data class Distance(val value: Double) {
    override fun toString(): String {
        return "%.4f".format(value)
    }
}

fun distance(location1: Location, location2: Location): Distance {
    val (x1, y1) = location1
    val (x2, y2) = location2

    val value = sqrt((x1 - x2).pow(2) + (y1 - y2).pow(2))
    return Distance(value)
}

File: lib/Location.py
'''
    Chartesian coordinate system with Euclidian distance
'''
from math import sqrt

class Location:

    ''' From array of strings
    '''
    def __init__(self, coordinates):
        try:
            self.__coordinates = []
            for c in coordinates:
                self.__coordinates.append( float(c) )
            self.__dim = len(self.__coordinates)
        except Exception as e :
            raise Exception("Coordinates format does not match!")


    ''' Euclidian distance
            rounded to 4 decimals
    '''
    def dist_to(self, b):
        ret = sqrt( sum( [ abs(self.__coordinates[i] - b.__coordinates[i])**2 for i in range(self.__dim) ] ) )
        ret = round(ret, 4)
        return ret
